### PR TITLE
refactor making util funcs more generic

### LINKED ISSUE #162: refactor: move controller funcs to exported funcs in util for more generic use with tests
rather than using controller for the client, we should pass the client to the func in an exported manner

### PR DESCRIPTION
**Pull request by screeley44**
fixes #162

### CODE REVIEW

File: pkg/controller/util.go [hunk 1]
```diff
@@ -36,6 +38,25 @@ func (c *Controller) pvcFromKey(key interface{}) (*v1.PersistentVolumeClaim, err
...
+		return nil, errors.New("key object not of type string")
+	}
+	obj, ok, err := informer.GetIndexer().GetByKey(keyString)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error getting key %q from cache", keyString)
+	}
+	if !ok {
+		return nil, errors.Errorf("key %q not found in cache", keyString)
+	}
+	pvc, ok := obj.(*v1.PersistentVolumeClaim)
```
Comment (copejon) [inline pkg/controller/util.go]: this func and podFromKey2 are the same down to this point in each.  could probably reduce some duplicate code

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -130,7 +130,8 @@ func (c *Controller) ProcessNextPodItem() bool {
 		return false
 	}
 	defer c.podQueue.Done(key)
-	pod, err := c.podFromKey(key)
+	//pod, err := c.podFromKey(key)
+	pod, err := podFromKey(c.podInformer, key)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: do we plan on any unit tests calling `podFromKey` or `pvcFromKey`? If so then they need to be exported. But my guess is no in which case they can remain methods.

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -165,11 +166,13 @@ func (c *Controller) processPodItem(pod *v1.Pod) error {
 		return errors.Errorf("Pod does not contain volume %q", DataVolName)
 	}
 	glog.V(Vdebug).Infof("processPodItem: Getting PVC object for key %q", pvcKey)
-	pvc, err := c.pvcFromKey(pvcKey)
+	//pvc, err := c.pvcFromKey(pvcKey)
+	pvc, err := pvcFromKey(c.pvcInformer, pvcKey)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: see above comment

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -187,13 +190,14 @@ func (c *Controller) ProcessNextPvcItem() bool {
 	}
 	defer c.pvcQueue.Done(key)
 
-	pvc, err := c.pvcFromKey(key)
+	pvc, err := pvcFromKey(c.pvcInformer, key)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: ditto

File: pkg/controller/controller.go [hunk 5]
```diff
@@ -187,13 +190,14 @@ func (c *Controller) ProcessNextPvcItem() bool {
...
 	if err != nil || pvc == nil {
 		return c.forgetKey(key, fmt.Sprintf("ProcessNextPvcItem: error converting key %q to pvc: %v", key, err))
 	}
 
 	// filter pvc and decide if the importer pod should be created
-	createPod, _ := c.checkPVC(pvc, false)
+	//createPod, _ := c.checkPVC(pvc, false)
+	createPod, _ := checkPVC(c.clientset, pvc, false)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: I'm not sure of the reason to convert a method to a func and not export it?  I don't know if we expect a unit test to test `checkPVC` but I guess there's a way to test private funcs by appending _Test_ to the func name...

**Comment by screeley44** [comment]
Closing this in favor of #168

File: pkg/controller/controller.go [hunk 6]
```diff
@@ -130,7 +130,8 @@ func (c *Controller) ProcessNextPodItem() bool {
 		return false
 	}
 	defer c.podQueue.Done(key)
-	pod, err := c.podFromKey(key)
+	//pod, err := c.podFromKey(key)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: delete comment

File: pkg/controller/controller.go (hunk 2 above)
Comment (jeffvance) [inline pkg/controller/controller.go]: keep this a method since it take an informer

File: pkg/controller/controller.go (hunk 4 above)
Comment (jeffvance) [inline pkg/controller/controller.go]: ditto on staying a method

**Pull request closed without merging**